integration-tests/src/test/java/com/datastax/oss/driver/mapper/SelectOtherClausesIT.java

```diff
@@ -0,0 +1,289 @@
+
+    for (String query :
+        ImmutableList.of("CREATE TABLE simple (k int, cc int, v int, PRIMARY KEY (k, cc))")) {
+      session.execute(
+          SimpleStatement.builder(query).setExecutionProfile(sessionRule.slowProfile()).build());
+    }
+
+    TestMapper mapper = TestMapper.builder(session).build();
+    dao = mapper.simpleDao(sessionRule.keyspace());
+
+    for (int k = 0; k < 2; k++) {
+      for (int cc = 0; cc < 10; cc++) {
+        dao.insert(new Simple(k, cc, 1));
+      }
+    }
+  }
+
+  @Test
+  public void should_select_with_limit() {
+    PagingIterable<Simple> elements = dao.selectWithLimit(10);
+    assertThat(elements.isFullyFetched()).isTrue();
+    assertThat(elements.getAvailableWithoutFetching()).isEqualTo(10);
+
+    elements = dao.selectWithLimit(0, 5);
+    assertThat(elements.isFullyFetched()).isTrue();
+    assertThat(elements.getAvailableWithoutFetching()).isEqualTo(5);
+
+    elements = dao.selectWithLimit(0, 0, 1);
+    assertThat(elements.isFullyFetched()).isTrue();
+    assertThat(elements.getAvailableWithoutFetching()).isEqualTo(1);
+  }
+
+  @Test
+  public void should_select_with_per_partition_limit() {
+    PagingIterable<Simple> elements = dao.selectWithPerPartitionLimit(5);
+    assertThat(elements.isFullyFetched()).isTrue();
+    assertThat(elements.getAvailableWithoutFetching()).isEqualTo(10);
+
+    Map<Integer, Integer> elementCountPerPartition = new HashMap<>();
+    for (Simple element : elements) {
+      elementCountPerPartition.compute(element.getK(), (k, v) -> (v == null) ? 1 : v + 1);
+    }
+    assertThat(elementCountPerPartition).hasSize(2).containsEntry(0, 5).containsEntry(1, 5);
+  }
+
+  @Test
+  public void should_select_with_order_by() {
+    PagingIterable<Simple> elements = dao.selectByCcDesc(0);
+    int previousCc = Integer.MAX_VALUE;
+    for (Simple element : elements) {
+      assertThat(element.getCc()).isLessThan(previousCc);
+      previousCc = element.getCc();
+    }
+  }
+
+  @Test
+  public void should_select_with_group_by() {
+    PagingIterable<Sum> sums = dao.selectSumByK();
+    assertThat(sums.all()).hasSize(2).containsOnly(new Sum(0, 10), new Sum(1, 10));
+  }
+
+  @Test
+  public void should_select_with_allow_filtering() {
+    PagingIterable<Simple> elements = dao.selectByCc(1);
+    assertThat(elements.all()).hasSize(2).containsOnly(new Simple(0, 1, 1), new Simple(1, 1, 1));
+  }
+
+  @Mapper
+  public interface TestMapper {
+    @DaoFactory
+    SimpleDao simpleDao(@DaoKeyspace CqlIdentifier keyspace);
+
+    static MapperBuilder<TestMapper> builder(CqlSession session) {
+      return new SelectOtherClausesIT_TestMapperBuilder(session);
+    }
+  }
+
+  @Dao
+  public interface SimpleDao {
+    @Insert
+    void insert(Simple simple);
+
+    @Select(limit = ":l")
+    PagingIterable<Simple> selectWithLimit(@CqlName("l") int l);
+
+    @Select(limit = ":l")
+    PagingIterable<Simple> selectWithLimit(int k, @CqlName("l") int l);
```

Comment:
These are examples of the new rule "if the method doesn't take the full primary key, the other parameters must be annotated with `@CqlName`". This only applies to `@Select`.